crates/dbsp/src/trace/consolidation/tests/mod.rs

```diff
@@ -158,42 +162,137 @@ fn consolidate_paired_slices_corpus() {
     assert_eq!(&values[..length], &[1]);
 }
 
+/// Checks that the given vectors are equivalent, even if their sorting
+/// algorithms have different orderings for identical keys
+#[track_caller]
+fn assert_sorted_eq<T, U>(
+    expected: &mut Vec<(T, U)>,
+    keys: &[T],
+    values: &[U],
+    buffer: &mut Vec<(T, U)>,
+) where
+    T: Clone + Ord + Debug,
+    U: Clone + Ord + Debug,
+{
+    assert_eq!(keys.len(), values.len());
+    assert_eq!(expected.len(), keys.len());
+
+    buffer.clear();
+    buffer.reserve(keys.len());
+    buffer.extend(keys.iter().cloned().zip(values.iter().cloned()));
+
+    for ((idx, (key, value)), (_, (next_key, next_value))) in
+        buffer.iter().enumerate().tuple_windows()
+    {
+        if key > next_key {
+            panic!("tuple at index {idx} isn't sorted: ({key:?}, {value:?}) > ({next_key:?}, {next_value:?})");
+        }
+    }
+
+    buffer.sort();
+    expected.sort();
+
+    for ((key, value), (expected_key, expected_value)) in buffer.iter().zip(expected) {
```

Comment:
Isn't this just `assert_eq(buffer, expected)`?

Reply:
Yes, but it gives a little more detail by telling what the actual incorrect key or value is